vmm/src/lib.rs

```diff
@@ -743,8 +750,44 @@ impl Vmm {
         socket
             .read_exact(&mut data)
             .map_err(MigratableError::MigrateSocket)?;
-        let config: VmConfig = serde_json::from_slice(&data).map_err(|e| {
-            MigratableError::MigrateReceive(anyhow!("Error deserialising config: {}", e))
+
+        let vm_migration_config: VmMigrationConfig =
+            serde_json::from_slice(&data).map_err(|e| {
+                MigratableError::MigrateReceive(anyhow!("Error deserialising config: {}", e))
+            })?;
+
+        #[cfg(all(feature = "kvm", target_arch = "x86_64"))]
+        // We check the `CPUID` compatibility of between the source vm and destination, which is
+        // mostly about feature compatibility and "topology/sgx" leaves are not relevant.
```

Comment:
Why do you say SGX leaves are not relevant? If you migrate to a destination without the SGX support, there's no way your migration will succeed.

Reply:
Given we are not supporting migration with `SGX` and the `SGX` feature bits from leaf 0x7 are already checked, we decide not to check SGX leaf 0x12 for now.